src/CollisionHandler.cpp

```diff
@@ -1,76 +1,109 @@
+#include "Quadtree.h"
 #include "CollisionHandler.h"
+#include "Marine.h"
+#include <iostream>
+
+CollisionHandler::CollisionHandler() : quadtreeMov(0, {0,0,2000,2000}), quadtreePro(0, {0,0,2000,2000}), 
+        quadtreeDam(0, {0,0,2000,2000}),quadtreePickUp(0, {0,0,2000,2000}) {
 
-CollisionHandler::CollisionHandler() {
-    quadtreeMov = new Quadtree(0, {0,0,2000,2000});
-    quadtreePro = new Quadtree(0, {0,0,2000,2000});
-    quadtreeDam = new Quadtree(0, {0,0,2000,2000});
-    quadtreePickUp = new Quadtree(0, {0,0,2000,2000});
 }
 
-CollisionHandler::~CollisionHandler() {
-    delete quadtreeMov;
-    delete quadtreePro;
-    delete quadtreeDam;
-    delete quadtreePickUp;
+CollisionHandler& CollisionHandler::operator=(const CollisionHandler& handle) {
+    quadtreeMov = handle.quadtreeMov;
+    quadtreePro = handle.quadtreePro;
+    quadtreeDam = handle.quadtreeDam;
+    quadtreePickUp = handle.quadtreePickUp;
+    return *this;
 }
 
 // Check for projectile collisions, return object it hits
-HitBox* CollisionHandler::detectDamageCollision(HitBox* hb) {
-    std::vector<HitBox*> returnObjects;
-    returnObjects = quadtreeDam->retrieve(returnObjects, hb);
-      for (unsigned int x = 0; x < returnObjects.size(); x++) {
-        if (returnObjects.at(x) != nullptr && hb->attached != returnObjects.at(x)->attached) {
-            if (SDL_HasIntersection(&hb->getRect(), &returnObjects.at(x)->getRect()) &&
-                !(hb->isPlayerFriendly() && returnObjects.at(x)->isPlayerFriendly()) ) {
-                return returnObjects.at(x);
-            }
+HitBox *CollisionHandler::detectDamageCollision(const HitBox *hb) {
+    std::vector<HitBox *> returnObjects;
+    returnObjects = quadtreeDam.retrieve(hb);
+    for (unsigned int x = 0; x < returnObjects.size(); x++) {
+        if (returnObjects.at(x) != nullptr && hb->attached != returnObjects.at(x)->attached 
```

Comment:
at() performs runtime bounds checking and throws exception on out-of-bounds.
If this is not the desired behaviour, and if the code doesn't not break when changing it to to operator[], please change.